src/components/Simulator.js

```diff
@@ -131,8 +132,17 @@ function Simulator() {
       <input type="number" min={0} value={numRuns} onChange={(event) => {
         setNumRuns(parseInt(event.target.value));
       }} />
+      Run Progress: {`${runsCompleted}/${numRuns}`}
       <br />
-      <button onClick={onRunMany}>
+      <button onClick={
+          () => onRunMany(
+            (runsCompleted) => {
+              if (runsCompleted % 10 === 0 || runsCompleted === numRuns) {
```

Comment:
Changing state on every run result didn't seem great, so i just have it update every 10.